src/main/java/com/labsynch/labseer/service/BulkLoadServiceImpl.java

```diff
@@ -382,23 +401,57 @@ public BulkLoadRegisterSDFResponseDTO registerSdf(BulkLoadRegisterSDFRequestDTO
 			CmpdRegSDFWriter registeredMolExporter = sdfWriterFactory.getCmpdRegSDFWriter(registeredSDFName);
 			HashMap<String, Integer> allAliasMaps = new HashMap<String, Integer>();
 
-			CmpdRegMolecule mol = null;
+			// CmpdRegMolecule mol = null;
 			int numRecordsRead = 0;
 			int numNewParentsLoaded = 0;
 			int numNewLotsOldParentsLoaded = 0;
-
-			while ((mol = molReader.readNextMol()) != null) {
-				numRecordsRead++;
-				try{
-					Boolean isNewParent = registerMol(mol, bulkLoadFile, mappings, errorCSVOutStream, registeredCSVOutStream, errorMolExporter, registeredMolExporter, allAliasMaps, numRecordsRead, results, validate, chemist, registerRequestDTO, dryRunCompound, startTime);
-					if(isNewParent != null) {
-						if (isNewParent)
-							numNewParentsLoaded++;
-						else
-							numNewLotsOldParentsLoaded++;
+			boolean done = false;
+
+			//TODO try to refactor this into a chunked reader with a clean iterator
+			int batchSize = propertiesUtilService.getStandardizationBatchSize();
+			while (!done) {
+				// Read the next set of molecules from the SDF
+				Collection<CmpdRegMolecule> batchOfMols = molReader.readNextMols(batchSize);
+				if (batchOfMols.size() == 0){
+					done = true;
+					break;
+				}
+				// Process structures in bulk
+				HashMap<String, String> origMolStructures = new LinkedHashMap<String, String>();
+				HashMap<String, CmpdRegMolecule> origMols = new LinkedHashMap<String, CmpdRegMolecule>();
```

Comment:
Wish I had looked up LinkedHashMap when refactoring for BBChem originally.  Nice to maintain order.